nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/nbio/ParseProtocol.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.nosqlbench.nb.api.nbio;
+
+import io.nosqlbench.nb.api.nbio.NBIO;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ParseProtocol {
+
+    private List<String> protocols;
+    private String path;
+
+    public ParseProtocol(String filepath) {
+        protocols = new ArrayList<>();
+
+        if (filepath.startsWith("http")) {
+            path = filepath;
+            protocols.add("all");
+        } else {
+            String[] parts = filepath.split(":");
+
+            if (parts.length < 2) {
+                path = filepath;
+                protocols.add("all");
+            } else {
```

Comment:
In some cases, I think there may be a ':' used as a suffix or qualifier, as in atfiles. There was also a previous convention used which would allow users to specify which version of yaml structure to load with `:v1` and `:v2` suffixes, but this _should_ be long gone.

I would suggest limiting the split to two terms (split method supports this), and ensuring that the first word is one from a specific set of accepted protocols, preferrably from an enum.

Reply:
I'll make this a more linear scan ...

Reply:
Changed to enums